src/runtime/composables/useDirectusItems.ts

```diff
@@ -1,87 +1,168 @@
-import {
+import type {
+  DirectusCollections,
   DirectusItemRequest,
   DirectusItemCreation,
   DirectusItemDeletion,
-  DirectusItemUpdate
+  DirectusItemUpdate,
+  DirectusQueryParamsMeta
 } from '../types'
+
 import { useDirectus } from './useDirectus'
 
-export const useDirectusItems = () => {
+export const useDirectusItems = <Collections extends DirectusCollections>() => {
   const directus = useDirectus()
 
-  const getItems = async <T>(data: DirectusItemRequest): Promise<T> => {
-    if (data.params?.filter) {
+  async function getItems <
+    C extends keyof Collections,
+    D extends DirectusItemRequest<Collections>[C] & {
+      params: {
+        // TODO: Somehow, 'meta: undefined | null' still results in this overload.
+        meta: NonNullable<D['params']['meta']>;
```

Comment:
If you want to remove `undefined` from possibles values, uses `-?`

```suggestion
        meta-?: NonNullable<D['params']['meta']>;
```

Reply:
Hmm, that doesn't seem like valid TS syntax. I think the `-?` [only works on mapped types](https://www.typescriptlang.org/docs/handbook/2/mapped-types.html#mapping-modifiers). But I actually cannot reproduce the issue that is described in the TODO. When providing `'meta': undefined` or `'meta': null`, it does not return this overload anymore. Not sure what changed in the meantime, maybe a TS update?

Reply:
I think he meant `meta?: NonNullable<D['params']['meta']>;`